core/components/ajaxupload/src/AjaxUpload.php

```diff
@@ -427,6 +428,16 @@ public function saveUploads($target, $clearQueue = false, $allowOverwrite = true
         }
         foreach ($this->session[$this->getOption('uid')] as $fileId => &$fileInfo) {
             if (file_exists($fileInfo['path'] . $fileInfo['uniqueName'])) {
+                if ($sanitizeFilename) {
+                    $pathinfo = pathinfo($fileInfo['originalName']);
+
+                    // Replace all spaces and punctuation characters with -
+                    $fileName = str_replace(['/',"'",'"','(',')',';','>','<',',','.','?','!'], '-', $pathinfo['filename']);
+                    $fileName = $this->modx->filterPathSegment($fileName);
+
```

Comment:
@Jako Replacing punctuation with regex proved to be a little tricky. Something like `/[[:punct:]]/` apparently removes some accented characters too, so I settled now on manually replacing punctuation characters. Maybe further sanitizing can be handled with transliteration, if needed?

Manual replacement is also done first now, so that filterPathSegment can mop up any double hyphens that might have come out.